scripts/helm/publish-chart-yaml.sh

```diff
@@ -10,15 +10,91 @@
 
+# Write output to error output stream.
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
 die()
 {
   local _return="${2:-1}"
-  echo "$1" >&2
+  echo_stderr "$1"
   exit "${_return}"
 }
 
 set -euo pipefail
 
+# This uses the existing remote refs for the openebs/mayastor-extensions repo to find the latest 'release/x.y' branch.
+# Requires a 'git fetch origin' (origin being the remote entry for openebs/mayastor-extensions) or equivalent, if not
+# done already.
+latest_release_branch() {
+  cd "$ROOTDIR"
+
+  local remote_url=""
+  local openebs_remote=""
+  for remote in $(git remote show); do
+    remote_url=$(git ls-remote --get-url $remote)
+    if [[ "$remote_url" =~ ([\/:]openebs\/mayastor-extensions)$ ]]; then
+      openebs_remote=$remote
+      break
+    fi
+  done
+  if [ -z "$openebs_remote" ]; then
+    die "failed to find git remote pointing to 'openebs/mayastor-extensions'"
+  fi
+  # The latest release branch name is required for generating the helm chart version/appVersion
+  # for the 'main' branch only.
+  # The 'git branch' command in the below lines checks remote refs for release/x.y branch entries.
+  # Because the 'main' branch is not a significant branch for a user/contributor, this approach towards
+  # finding the latest release branch assumes that this script is used when the 'openebs/mayastor-extensions'
+  # repo is present amongst git remote refs. This happens automatically when the 'openebs/mayastor-extensions'
+  # repo is cloned, and not a user/contributor's fork.
+  local latest_release_branch=$(git branch \
+    --all \
+    --list "*${openebs_remote}/release/*.*" \
+    --format '%(refname:short)' \
+    --sort 'refname' \
+    | tail -n 1)
+
+  cd - >/dev/null
+
+  echo "${latest_release_branch#*$openebs_remote/}"
+}
+
+main_branch_version_from_release_branch() {
+  local release_branch=$1
+
+  # Validate release branch name.
+  if ! [[ "$release_branch" =~ ^(release\/[0-9]+\.[0-9]+)$ ]]; then
+    die "'$release_branch' is not a valid release branch"
+  fi
+
+  # It is possible that this version isn't released, and may never see an actual release.
+  # However, it is the latest possible release, and no newer version of a higher semver
+  # values likely exists on that repo.
+  # Adds a '.0' at the end to make the version semver compliant.
+  local latest_version="${release_branch#*release/}".0
+
+  local bumped_latest=""
+  # Bump minor by default.
+  if [ -n "$BUMP_MAJOR_FOR_MAIN" ]; then
+    bumped_latest=$(semver bump major "$latest_version")
+  else
+    bumped_latest=$(semver bump minor "$latest_version")
+  fi
+
+  # Date-time appended to main tag, if present. Else, defaulting to 'main'.
+  local penultimate_member=""
+  if [ -n "$DATE_TIME" ]; then
+    penultimate_member=$DATE_TIME
+  else
+    penultimate_member="main"
+    echo_stderr "WARNING: No input timestamp for main branch helm chart, using 'main' instead"
```

Comment:
We don't. I've put this in, so that the `--check-chart "main"` flag works without needing the `--date-time $(date +%Y-%m-%d-%H-%M-%S)` option, just to keep with the usage pattern from all other branches.

It doesn't let you generate a chart with `main` written in it instead of the timestamp (check line 328).

Should I get rid of it?

Reply:
So what would the app version look like here at the end?

Reply:
APP_TAG: <bumped-latest>-0-develop-unstable-main-0

This will be printed too. This is not allowed when `--develop-to-main` is also specified.

Reply:
oh I see got it, up to you, take it or leave it.

Reply:
I'd want to leave it in.